src/orchard/meta.clj

```diff
@@ -268,11 +268,15 @@
      (meta ns)
      {:ns (ns-name ns)
       :name (ns-name ns)
-      :file (-> (ns-publics ns)
-                first
-                second
-                var-meta
-                :file)
+      :file (or 
+             (-> (ns-publics ns)
+                 first
+                 second
+                 var-meta
+                 :file)
+             (->
+              (ns/canonical-source ns)
```

Comment:
I guessing when you try this manually it actually works (instead of returning nil), right?

Reply:
Yes, `(ns-meta 'orchard.test-no-defs)` returns the right thing. And the `:clj` ”branch” of the test also passes. But the `:cljs` part of the test does not.

Reply:
Seems `canonical-source` doesn't work with `cljs` files:

```
(defn canonical-source
  "Returns the URL of the source file for the namespace object or symbol,
  according to the canonical naming convention, if present on the classpath."
  [ns]
  (let [path (-> (str ns)
                 (str/replace "-" "_")
                 (str/replace "." "/"))]
    (or (io/resource (str path ".clj"))
        (io/resource (str path ".cljc")))))
```

Reply:
Good catch! This would have made me fail fix https://github.com/BetterThanTomorrow/calva/issues/427 even when we find out why the cljs tests fail. 😄